### PR TITLE
web: `Definition` and `History` sections Code Intel panel redesign 

### PR DESCRIPTION
**Pull request by valerybugakov**
## Description

Definition and History sections Code Intel panel redesign: https://github.com/sourcegraph/sourcegraph/issues/21045.
[Figma mockups](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Design-Refresh-Systemization-source-of-truth?node-id=3008%3A501).

## Changes

- Added redesign styles for Definition and History sections.
- Updated snapshots accordingly.

### History section

Screenshot 2021-06-02 at 14 53 24

### Definition section

Screenshot 2021-06-02 at 14 53 36

## Notes

1. Alignment of the file icon in the History section is different from the design. It can be changed easily. I want to double-check that we don't want to align it with the close icon in the header row. See [thread](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw?node-id=3008:501#84632333).

<p align="center">
  Screenshot 2021-06-02 at 13 52 50
&nbsp; &nbsp;
  Screenshot 2021-06-02 at 13 49 08
</p>

2. Left padding in the History section is different from the design according to this [thread](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw?node-id=3008:501#84466972).

### CODE REVIEW

File: client/web/src/repo/commits/GitCommitNode.tsx [hunk 1]
```diff
@@ -73,10 +76,12 @@ export const GitCommitNode: React.FunctionComponent<GitCommitNodeProps> = ({
             compact={Boolean(compact)}
         />
     )
+
+    const MessageWrapper = isRedesignEnabled && compact ? 'small' : React.Fragment
```
Comment (tjkandala) [inline client/web/src/repo/commits/GitCommitNode.tsx:80]: Clean 😎️

**Review by tjkandala — APPROVED** [review]
Code LGTM

File: client/web/src/components/FilteredConnection/FilteredConnection.scss [hunk 2]
```diff
@@ -33,7 +33,7 @@ $compact-summary-min-height: 2.75rem;
         padding: var(--popover-item-padding-v) var(--popover-item-padding-h);
         flex: 0 0;
 
-        .theme-redesign & {
+        :where(.theme-redesign) & {
```
[the remark this answers is not in the document]
Comment (valerybugakov) [reply client/web/src/components/FilteredConnection/FilteredConnection.scss:36]: `:where()` is added in a couple of places in this PR to lower the specificity of the `.theme-redesign` selector to change CSS rules via a custom class name. For this particular case, it's `loaderClassName` that was added to the `FilteredConnection` component.

File: client/web/src/repo/RepoRevisionSidebarCommits.tsx [hunk 3]
```diff
@@ -21,22 +22,24 @@ import { replaceRevisionInURL } from '../util/url'
...
+import styles from './RepoRevisionSidebarCommits.module.scss'
 
 interface CommitNodeProps {
     node: GitCommitFields
     location: H.Location
 }
 
 const CommitNode: React.FunctionComponent<CommitNodeProps> = ({ node, location }) => (
-    <li className="list-group-item p-0">
+    <li className={classNames(styles.repoRevisionSidebarCommitsCommitContainer, 'list-group-item p-0')}>
```
[the remark this answers is not in the document]
Comment (valerybugakov) [reply client/web/src/repo/RepoRevisionSidebarCommits.tsx:33]: I've seen both approaches being used in the codebase. We need to add guidelines to our docs about that. 
I don't see any issues going without BEM. Will push the update 👍 

cc @sourcegraph/frontend-devs

[the remark this answers is not in the document]
Comment (valerybugakov) [reply client/web/src/repo/RepoRevisionSidebarCommits.tsx]: Added a note in [this PR](https://github.com/sourcegraph/sourcegraph/pull/21681). Let's discuss it there.

**Review by eseliger — APPROVED** [review]
Nice 👌

**Pull request merged**